src/realm/object-store/sync/app.cpp

```diff
@@ -1235,36 +1203,12 @@ void App::do_request(std::unique_ptr<Request>&& request, IntermediateCompletion&
     // If location info has already been updated, then send the request directly
     auto& request_ref = *request;
     m_config.transport->send_request_to_server(
-        request_ref, [self = shared_from_this(), completion = std::move(completion),
-                      request = std::move(request)](const Response& response) mutable {
-            self->check_for_redirect_response(std::move(request), response, std::move(completion));
+        request_ref,
+        [completion = std::move(completion), request = std::move(request)](const Response& response) mutable {
```

Comment:
do we need to capture self here to keep `App` alive for the duration of the request,  or is the contract here that the completion is responsible for keeping the `App` alive?

Reply:
The completion callback provided to `do_request()` captures an `App` shared pointer, which will ensure the app sticks around until the callback is called.
The only exception is when the original request is re-sent after successfully refreshing the user access token after the request sent by `do_authenticated_request()` initially failed.
I will update this call and add a statement to `do_request()` regarding ensuring the callback contains an `App` shared ptr instance in the captures.